### DISCUSSION TITLE
Reproduce ipynb code lines in a Jupyter Book

### DISCUSSION THREAD

**Discussion by Jdbeck66** [executablebooks/meta#767] [Q&A]
Is there any way to include line numbering from code blocks in a Jupyter notebook within a Jupyter Book.  I have line numbers shown in the select notebook cells, but when I compile the Jupyter Book, they are not reflected in the html. See below.
![Screen Shot 2022-06-30 at 2 54 33 PM](https://user-images.githubusercontent.com/11720651/176776648-b75a7b82-170d-4512-99cb-2c1aa16ba996.png)
![Screen Shot 2022-06-30 at 2 54 47 PM](https://user-images.githubusercontent.com/11720651/176776667-0e5545bc-683f-4b19-8279-6ec3168600fc.png)

This would be helpful as I'd like to refer to specific lines for documentation purposes.

Thanks for any help.

**Comment by choldgraf** [comment]
I believe that this is going to be possible when Jupyter Book updates to the latest MyST-NB, but this has not happened yet, so maybe keep an eye out for this one. Here is the relevant docs in MyST-NB: https://myst-nb.readthedocs.io/en/latest/render/format_code_cells.html#number-source-lines

**Comment by Jdbeck66** [comment]
Is that only for markdown files or will .ipynb files that have line
numbering within a cell also render in the html build?

Thanks for your response.

On Thu, Jun 30, 2022 at 10:07 PM Chris Holdgraf ***@***.***>
wrote:

> I believe that this is going to be possible when Jupyter Book updates to
> the latest MyST-NB, but this has not happened yet, so maybe keep an eye out
> for this one. Here is the relevant docs in MyST-NB:
> https://myst-nb.readthedocs.io/en/latest/render/format_code_cells.html#number-source-lines
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/executablebooks/discussions/767#discussioncomment-3060963>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACZNPSZGKC4CHECHWGHSBQDVRZVGHANCNFSM52KUQJKA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 
Jim Beck
Ph.D. Candidate - Computing, Computational Math, Science, and Engineering
*Boise State University*

**Reply by choldgraf** [reply]
it should be for both cases, I believe...markdown notebooks via myst aren't treated differently from `ipynb` files once they're parsed
